templates/deployments.yaml

```diff
@@ -106,6 +104,141 @@ spec:
 {{ toYaml . | indent 8 }}
     {{- end }}
 
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-st2chatops{{ template "enterpriseSuffix" . }}
+  labels:
+    app: st2chatops
+    tier: backend
+    vendor: stackstorm
+    support: {{ template "supportMethod" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app: st2chatops
+      support: {{ template "supportMethod" . }}
+      release: {{ .Release.Name }}
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: st2chatops
+        tier: backend
+        vendor: stackstorm
+        support: {{ template "supportMethod" . }}
+        chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmaps_st2-conf.yaml") . | sha256sum }}
+        checksum/chatops: {{ include (print $.Template.BasePath "/configmaps_chatops.yaml") . | sha256sum }}
+        checksum/packs: {{ include (print $.Template.BasePath "/configmaps_packs.yaml") . | sha256sum }}
+        checksum/auth: {{ include (print $.Template.BasePath "/secrets_st2auth.yaml") . | sha256sum }}
+        checksum/ssh: {{ include (print $.Template.BasePath "/secrets_ssh.yaml") . | sha256sum }}
+    spec:
+      {{- if .Values.enterprise.enabled }}
+      imagePullSecrets:
+      - name: {{ .Release.Name }}-st2-license
+      {{- end }}
+      initContainers:
+      {{- if .Values.st2.packs.image.repository }}
+      # Merge packs and virtualenvs from st2actionrunner with those from the st2.packs image
+      # Can be used for chatops pack if system pack comes without it.
```

Comment:
I'm under impression that access to pack content is not needed for `st2chatops/hubot` itself, as it's something that registered in st2 core.

I think all the section related to custom pack content could be removed from `st2chatops` pod, but can you describe the edge case with the system pack?

Reply:
The only edge case I can think of is when we do not have `chatops` pack installed with other system packs. (not sure if that's even possible) That case can easily render `st2chatops` useless. not to think what will happen to `notifications`

I was originally thinking of a condition where `chatops` pack is not installed and tried to find a way to overcome that scenario.